src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -423,6 +439,19 @@ public Optional<ImmutableMap<Version, String>> getYankedVersions(
     }
   }
 
+  @Override
+  public boolean shouldFetchYankedVersions(ModuleKey selectedModuleKey) {
+    // If the source.json hash is known, this module has been selected before when selection
+    // succeeded, which means that either:
+    // * it wasn't yanked at that point in time and any successful selection since then has not seen
```

Comment:
I'm a bit confused by the wording "this module has been selected before" and "any successful selection since then has not seen a higher module version". If the source.json hash is known, isn't that because this module version was selected *exactly* last time? If it wasn't selected "last time", we wouldn't even have it in the lockfile.

Unless you're referring to the case where git merges the lockfile without our knowledges (i.e. the last "author" of the lockfile was git, not Bazel), in which case -- what's preventing git from merging in the source.json hash of foo@1.2 when foo@1.1 is already in there?

Reply:
I edited the comment, please let me know whether this is clearer now.